python/ray/data/datasource/sql_datasource.py

```diff
@@ -29,6 +30,21 @@ def __init__(self, connection_factory: Callable[[], Connection]):
     def create_reader(self, sql: str) -> "Reader":
         return _SQLReader(sql, self.connection_factory)
 
+    def write(
+        self,
+        blocks: Iterable[Block],
+        ctx: TaskContext,
+        sql: str,
+    ) -> WriteResult:
+        with _connect(self.connection_factory) as cursor:
+            for block in blocks:
+                block_accessor = BlockAccessor.for_block(block)
+                rows = list(block_accessor.iter_rows(public_row_format=True))
```

Comment:
split the rows in small batches? otherwise it may OOM for large blocks

Reply:
yeah that can be followup, let's see how @pcmoritz test goes first?

Reply:
Thanks for implementing this! I got around trying this out now, and unfortunately with postgres it is not writing anything to the DB for me, I wonder what is going wrong. I saw in the `_connect` method, you are doing a `rollback` for the transaction (since I assume this was for the read part), could that be the problem.

Reply:
The test has the same behavior btw:

```python
Python 3.9.15 (main, Nov  4 2022, 16:13:54) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sqlite3

In [2]: connection = sqlite3.connect("/tmp/db.sqlite")

In [3]: connection.cursor().execute("CREATE TABLE test(string, number)")
Out[3]: <sqlite3.Cursor at 0x7f7d8ffb8c70>

In [5]: import ray

In [6]: dataset = ray.data.from_items(
   ...:         [{"string": "spam", "number": 0}, {"string": "ham", "number": 1}]
   ...:     )

In [7]: dataset.write_sql(
   ...:         "INSERT INTO test VALUES(?, ?)", lambda: sqlite3.connect("/tmp/db.sqlite"))

In [8]: result = connection.cursor().execute("SELECT * FROM test ORDER BY number")

In [9]: result.fetchall()
Out[9]: []

In [10]: 
```

So somehow maybe the test is not running at the moment :)

Reply:
Oh, that's dumb. Forgot to add `assert` to my assertion (－‸ლ) https://github.com/ray-project/ray/blob/d6b76317400914ac944623747cbcf45633f05948/python/ray/data/tests/test_sql.py#L50

Reply:
Should be fixed now

Reply:
@raulchen added chunking as discussed offline